linera-chain/src/chain.rs

```diff
@@ -744,6 +745,8 @@ where
                 ChainError::InactiveChain(self.chain_id())
             );
         }
+        let mut oracle_records = oracle_records.map(Vec::into_iter);
```

Comment:
Seems to indicate that a single vector might be preferable?

Reply:
Why? In the iteration in this function we want to pass the oracle record (i.e. the _list_ of responses) into `execute_message` and `execute_operation` that applies that _that_ transaction.

_Inside_ transaction handling, in the runtime, we iterate over the individual responses.

The `map` here is an `Option::map`: If it's `None` that means there are no oracle records at all for this block because it's being executed for the first time, so the actual oracles have to be consulted.